server/src/main/java/org/elasticsearch/cluster/routing/allocation/allocator/DesiredBalanceShardsAllocator.java

```diff
@@ -251,6 +251,11 @@ public void onFailure(Exception e) {
             assert MasterService.isPublishFailureException(e) : e;
             onNoLongerMaster();
         }
+
+        @Override
+        public String toString() {
+            return "ReconcileDesiredBalanceTask[lastConvergedIndex=" + desiredBalance.lastConvergedIndex() + "]";
+        }
```

Comment:
NIT: may be this could be converted to a record?

Reply:
It's not a `static` class. I mean yes we could pass in a reference to the `DesiredBalanceShardsAllocator` explicitly, but I don't see the benefit. Also I don't want the `toString()` to contain the whole desired balance really, just the index is enough.